metadata-ingestion/src/datahub/ingestion/source/state/stale_entity_removal_handler.py

```diff
@@ -233,6 +245,21 @@ def _create_soft_delete_workunit(self, urn: str, type: str) -> MetadataWorkUnit:
         report.report_stale_entity_soft_deleted(urn)
         return wu
 
+    def add_urn_to_skip(self, urn: str) -> None:
+        # We previously had bugs where sources (e.g. dbt) would add non-primary entity urns
```

Comment:
We need to define this concept of "primary urn provider" more tightly

Reply:
i suspect we'll need to evolve this in the future. it's good enough for now and imo fairly clear, and we can rename/add docs once we start to encounter other usages of it